tools/c7n_oci/c7n_oci/resources/dns.py

```diff
@@ -0,0 +1,147 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import re  # noqa
+import copy  # noqa
+
+import oci.dns
+
+from c7n.filters import Filter, ValueFilter  # noqa
+from c7n.utils import type_schema
+from c7n_oci.actions.base import OCIBaseAction, RemoveTagBaseAction
+from c7n_oci.provider import resources
+from c7n_oci.query import QueryResourceManager
+
+log = logging.getLogger("custodian.oci.resources.dns")
+
+
+@resources.register("zone")
+class Zone(QueryResourceManager):
+    """Oracle Cloud Infrastructure Zone Resource
+
+    :example:
+
+    Returns all Zone resources in the tenancy
+
+    .. code-block:: yaml
+
+        policies:
+            - name: find-all-zone-resources
+              resource: oci.zone
+
+    """
+
+    class resource_type:
+        doc_groups = ["DNS"]
+        service = "oci.dns"
+        client = "DnsClient"
+        enum_spec = ("list_zones", "items[]", None)
+        extra_params = {"compartment_id"}
+        resource_type = "OCI.Dns/Zone"
+        id = "id"
+        name = "name"
+        search_resource_type = "customerdnszone"
+
+
+@Zone.action_registry.register("update-zone")
+class UpdateZone(OCIBaseAction):
+    """
+        Update zone Action
+
+        :example:
+
+        Updates the zone with the specified information.
+
+    Global secondary zones may have their external masters updated. For more information about secondary
+    zones, see [Manage DNS Service Zone](/iaas/Content/DNS/Tasks/managingdnszones.htm). When the zone name
+    is provided as a path parameter and `PRIVATE` is used for the scope query parameter then the viewId
+    query parameter is required.
+
+
+        Please refer to the Oracle Cloud Infrastructure Python SDK documentation for parameter details to this action
+        https://docs.oracle.com/en-us/iaas/tools/python/latest/api/dns/client/oci.dns.DnsClient.html#oci.dns.DnsClient.update_zone
+
+        .. code-block:: yaml
+
+            policies:
+                - name: perform-update-zone-action
+                  resource: oci.zone
+                  actions:
+                    - type: update-zone
+
+    """  # noqa
+
+    schema = type_schema("update-zone", params={"type": "object"}, rinherit=OCIBaseAction.schema)
+
+    def perform_action(self, resource):
+        client = self.manager.get_client()
+        params_dict = {}
+        params_model = {}
+        if self.data.get("params") and self.data.get("params").get("zone_name_or_id"):
```

Comment:
the parameters should come from the resource, not specified via action parameters. an action operates on a set of resources that have matched filters, this effectively is styled to bypass that behavior and perform the action against a statically defined resource for every resource that dynamically matches filters.